source/Api.cpp

```diff
@@ -71,7 +75,9 @@ namespace Aws
             }
 
             g_allocator = nullptr;
-            aws_s3_library_clean_up();
+            #ifdef ENABLE_S3_BUILT
+                aws_s3_library_clean_up();
```

Comment:
Similarly, `aws_s3_library_clean_up` cleanup several other libraries, including `aws_http_library_clean_up` and `aws_auth_library_clean_up`